MinecraftClient/ChatBot.cs

```diff
@@ -220,7 +220,19 @@ private McClient Handler
         /// <param name="locked"></param>
         /// <param name="iconcount"></param>
         public virtual void OnMapData(int mapid, byte scale, bool trackingposition, bool locked, int iconcount) { }
-        
+
+        /// <summary>
+        /// Called when tradeList is received from server
+        /// </summary>
+        /// <param name="windowID">Window ID</param>
+        /// <param name="size">Total Amount of trades the villager has.</param>
+        /// <param name="trades">List of trades.</param>
+        /// <param name="villagerLevel">The level the villager is.</param>
+        /// <param name="experience">The amount of experience the villager has.</param>
+        /// <param name="isRegularVillager">True if regular villagers and false if the wandering trader.</param>
+        /// <param name="canRestock">If the villager can restock his trades at a workstation, True for regular villagers and false for the wandering trader.</param>
+        public virtual void OnTradeList(int windowID, int size, List<Trade> trades, int villagerLevel, int experience, bool isRegularVillager, bool canRestock) { }
```

Comment:
How about removing the `int size` parameter as the size can be obtained from `List<Trade> trades`?

Reply:
Indeed. Size is not necessary because C# `List` has a `Count` attribute.

Additionally, since villager properties may change in newer Minecraft versions, adjusting the parameters in this ChatBot event would beak existing scripts. To make this event more future-proof, how about moving these to a `VillagerInfo` class? The method would look like this:

```C#
public virtual void OnTradeList(int windowID, List<Trade> trades, VillagerInfo villagerInfo) { }
```
```C#
public class VillagerInfo
{
    public int Level { get; set; }
    public int Experience { get; set; }
    public bool IsRegularVillager { get; set; }
    public bool CanRestock { get; set; }
}
```
This way, new fields can be added without breaking existing scripts, and existing getters can be adjusted if necessary.

Reply:
I can't figure out how to get the actual value into the set part. What I could do is add a dataType just like trade but that is probably not what you intended right?

Reply:
Assuming the class is the one presented above, you can access each field directly:
```C#
VillagerInfo villagerInfo = new VillagerInfo();
villagerInfo.Level = 1;
villagerInfo.Experience = 1;
villagerInfo.IsRegularVillager = true;
villagerInfo.CanRestock = true;
```
Or, if you want, even initialize all fields at once:
```C#
VillagerInfo villagerInfo = new VillagerInfo()
{
    Level = 1,
    Experience = 1,
    IsRegularVillager = true,
    CanRestock = true
};
```

Reply:
I made a new file for the class I don't know if that's the best way to do it.